skiko/src/iosMain/kotlin/org/jetbrains/skiko/SkikoUIView.kt

```diff
@@ -181,13 +182,26 @@ class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol,
         return _pointInside(skiaPoint, withEvent)
     }
 
+    private var touchesCount = 0
+        set(value) {
+            field = value
+
+            skiaLayer?.redrawer?.needsProactiveDisplayLink = value != 0
+        }
+
     override fun touchesBegan(touches: Set<*>, withEvent: UIEvent?) {
         super.touchesBegan(touches, withEvent)
+
+        touchesCount += touches.size
+
         sendTouchEventToSkikoView(withEvent!!, SkikoPointerEventKind.DOWN)
     }
 
     override fun touchesEnded(touches: Set<*>, withEvent: UIEvent?) {
         super.touchesEnded(touches, withEvent)
+
+        touchesCount -= touches.size
```

Comment:
We need to check if newly introduced variables can be accessed from multiple threads. If they can, it is better to declare them as Atomic to avoid race conditions

Reply:
@MatkovIvan 
Yes, it's safe. iOS guarantees lifecycle of touches intercepted by UIView. It begins and either ends or cancels.

@pjBooms 
Touch events on iOS are dispatched on RunLoop.main, which is bound to main thread, no data races.